proto/frontend/src/device_mgr.cpp

```diff
@@ -890,13 +890,16 @@ class DeviceMgrImp {
     RETURN_OK_STATUS();
   }
 
+  typedef
+  std::unordered_map<pi_indirect_handle_t, p4v1::ActionProfileActionSet>
+  action_profile_set_map;
+
   Status parse_action_entry(
       p4_id_t table_id,
       const pi_table_entry_t *pi_entry,
       p4v1::TableEntry *entry,
-      const std::unordered_map<
-        pi_indirect_handle_t,
-        p4v1::ActionProfileActionSet> &oneshot_map = {}) const {
```

Comment:
@antoninbas I am not able to build it with `oneshot_map = {}`. 

The error message is
```
proto/frontend/src/device_mgr.cpp:903:53: error: converting to 'const std::unordered_map<long unsigned int, p4::v1::ActionProfileActionSet>' from initializer list would use explicit constructor 'std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::unordered_map(std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::size_type, const hasher&, const key_equal&, const allocator_type&) [with _Key = long unsigned int; _Tp = p4::v1::ActionProfileActionSet; _Hash = std::hash<long unsigned int>; _Pred = std::equal_to<long unsigned int>; _Alloc = std::allocator<std::pair<const long unsigned int, p4::v1::ActionProfileActionSet> >; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::size_type = long unsigned int; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::hasher = std::hash<long unsigned int>; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::key_equal = std::equal_to<long unsigned int>; std::unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>::allocator_type = std::allocator<std::pair<const long unsigned int, p4::v1::ActionProfileActionSet> >]'
       p4v1::ActionProfileActionSet> &oneshot_map = {}) const {
```

Any idea?

Reply:
I guess the reason is that I am using c++11 ?
https://stackoverflow.com/questions/26947704/implicit-conversion-failure-from-initializer-list

Reply:
The Travis CI also uses c++11, but I think whether it builds ultimately depends on the compiler / libstdc++ version being used. More recent compiler versions will probably accept the code.
The change is fine but:
 * please use `using` and not `typedef` for consistency
 * please update the rest of the device_mgr code to use `action_profile_set_map` (e.g. here: https://github.com/p4lang/PI/blob/master/proto/frontend/src/device_mgr.cpp#L951)